QMLComponents/variableinfo.cpp

```diff
@@ -1,13 +1,24 @@
 #include "variableinfo.h"
 #include "jasptheme.h"
+#include "QQmlContext"
+#include "QTimer"
 
 VariableInfo* VariableInfo::_singleton = nullptr;
 
 VariableInfo::VariableInfo(VariableInfoProvider* providerInfo) :
 	QObject(providerInfo->providerModel()), _provider(providerInfo)
 {
 	if (_singleton == nullptr)
+	{
 		_singleton = this;
+		QTimer::singleShot(0, [&]() { _setDataSetInfoInContext(); });
```

Comment:
Why not set it from mainwindow loadQML?

Reply:
I find more logical that the QMLComponents module itself registers the VariableInfo/dataSetInfo object.
The QMLComponents module should be as independent as possible of its environment, only a tiny interface (VariableInfoProvider) defines what is needed from the environment.
Now it is clear that other environments than JASPDesktop should provide the QML root Context object to QMLComponents so that the module can work: the module is not dependent on the VariableInfo object being registered elsewhere.

Reply:
Ok cool!

Reply:
No sorry I still want to test it but I first want to build development